Sources/NIOHTTP2/HTTP2StreamChannel.swift

```diff
@@ -554,11 +554,19 @@ private extension HTTP2StreamChannel {
         self.pendingReads = CircularBuffer(initialCapacity: 0)
     }
 
-    /// DinitialCapacityreads to the channel.
+    /// Deliver all pending reads to the channel.
     private func deliverPendingReads() {
         assert(self.isActive)
         while self.pendingReads.count > 0 {
-            self.pipeline.fireChannelRead(NIOAny(self.pendingReads.removeFirst()))
+            let frame = self.pendingReads.removeFirst()
```

Comment:
For minor CoW purposes it would be best if we calculated the flow controlled length _first_. This would allow us to pass the frame into `fireChannelRead` and no longer own the data.

Reply:
Ah, good call!